src/manager.rs

```diff
@@ -135,10 +121,14 @@ impl<S: MutEventSubscriber> EventManager<S> {
         #[cfg(feature = "remote_endpoint")]
         let mut endpoint_event = None;
 
-        // Use the temporary, pre-allocated buffer to check ready events.
+        // TODO: implement an iterator for EpollWrapper to simplify the abstraction
         for ev_index in 0..event_count {
-            let event = self.ready_events[ev_index];
+            let event = self.epoll_context.ready_events[ev_index];
             let fd = event.fd();
+            if event.events() == 0 && fd == 0 {
```

Comment:
Can you add a comment here to why is this "softly" removed (i.e. we cannot actually remove elements from the ready_events while processing them because this would make the iterator go wild, instead we're just setting the removed entry to EpollEvent::default().